src/query/runner/ipa.rs

```diff
@@ -23,20 +23,18 @@ impl Runner {
         ctx: SemiHonestContext<'_>,
         field: FieldType,
         input: ByteArrStream,
-    ) -> Box<dyn ProtocolResult> {
-        match field {
+    ) -> Result<Box<dyn ProtocolResult>, Error> {
```

Comment:
Can/should this use `QueryResult`? (Even if yes, no need to change in this PR, because I have more changes coming to this code.)

Reply:
oh yea, my bad - missed it. Leaving it as is as suggested